cumulusci/tasks/package_2gp.py

```diff
@@ -0,0 +1,537 @@
+    minor = "minor"
+    patch = "patch"
+
+
+class PackageConfig(BaseModel):
+    name: str
+    description: str = ""
+    package_type: PackageTypeEnum
+    namespace: Optional[str]
+    branch: str = None
+    version_name: str
+    version_type: VersionTypeEnum = VersionTypeEnum.minor
+
+
+class CreatePackageVersion(BaseSalesforceApiTask):
+    """Creates a new second-generation package version.
+
+    If a package named ``package_name`` does not yet exist in the Dev Hub, it will be created.
+    """
+
+    api_version = "48.0"
+
+    task_options = {
+        "package_name": {"description": "Name of package"},
+        "package_type": {
+            "description": "Package type (Unlocked or Managed)",
+            "required": True,
+        },
+        "namespace": {"description": "Package namespace"},
+        "version_name": {"description": "Version name"},
+        "version_type": {
+            "description": "The part of the version number to increment. "
+            "Options are major, minor, patch.  Defaults to minor"
+        },
+        "dependency_org": {
+            "description": "The org name of the org to use for project dependencies lookup. If not provided, a scratch org will be created with the org name 2gp_dependencies."
+        },
+        "skip_validation": {
+            "description": "If true, skip validation of the package version. Default: false. "
+            "Skipping validation creates packages more quickly, but they cannot be promoted for release."
+        },
+    }
+
+    def _init_options(self, kwargs):
+        super()._init_options(kwargs)
+
+        self.package_config = PackageConfig(
+            name=self.options.get("name") or self.project_config.project__package__name,
+            package_type=self.options.get("package_type")
+            or self.project_config.project__package__type,
+            namespace=self.options.get("namespace")
+            or self.project_config.project__package__namespace,
+            branch=self.project_config.repo_branch,
+            version_name=self.options.get("version_name") or "Release",
+            version_type=self.options.get("version_type") or "minor",
+        )
+        self.options["skip_validation"] = process_bool_arg(
+            self.options.get("skip_validation", False)
+        )
+
+    def _run_task(self):
+        """Creates a new 2GP package version.
+
+        1. Create package if not found in Dev Hub.
+        2. Request creation of package version.
+        3. Wait for completion.
+        4. Collect package information as return values.
+        """
+        # find existing package in Dev Hub, or create one if necessary
+        self.package_id = self._get_or_create_package(self.package_config)
+        self.return_values["package_id"] = self.package_id
+
+        # submit request to create package version
+        package_zip_builder = MetadataPackageZipBuilder(
+            path=self.project_config.default_package_path,
```

Comment:
I think sfdx is working on a change so that multiple package directories get deployed in multiple metadata API operations in a well-defined order, rather than a combined one. I'm not sure how that will translate to package creation.